include/swift/Sema/ConstraintSystem.h

```diff
@@ -3906,7 +3874,7 @@ class ConstraintSystem {
     // Add this constraint to the constraint graph.
     CG.addConstraint(constraint);
 
-    if (isDebugMode() && getPhase() == ConstraintSystemPhase::Solving) {
+    if (isDebugMode() && solverState) {
```

Comment:
Strictly speaking this isn't the same thing since it means we'll now show unnecessary debug output when generating constraints for e.g a closure body. Alternatively we could just suppress debug output during constraint generation (by removing `DebugConstraints` from the options), or we could keep a "is generating constraints" bit for this. Any thoughts?